creating_images/guidelines.adoc

```diff
@@ -268,11 +268,56 @@ written to be a particular user ID, and then runs the image using that same user
 ID. This ensures the volume is only accessible to the appropriate container, but
 requires the image be able to run as an arbitrary user ID.
 
+Running a container with an arbitrary user ID also has the benefit of ensuring that a 
+process which is able to escape the container due to a vulnerability in the container 
+framework will not have specific user permissions on the host system.
+
 To accomplish this, directories that must be written to by processes in the
 image should be world-writable. In addition, the processes running in the
 container must not listen on privileged ports (ports below 1024).
 
-Lastly, the final *USER* declaration in the Dockerfile should specify the user
+Because the user ID of the container is generated dynamically, it will not have an
+associated entry in `/etc/passwd`.  This can cause a problem for some applications
+which expect to be able to look up their user ID.  One way to address this problem
+is to use https://cwrap.org/nss_wrapper.html[nss wrapper] and dynamically create
+a `passwd` file with the container's user ID as part of the image's start script:
+
+----
+export USER_ID=$(id -u)
+export GROUP_ID=$(id -g)
+envsubst < ${HOME}/passwd.template > ${HOME}/passwd
+export LD_PRELOAD=libnss_wrapper.so
+export NSS_WRAPPER_PASSWD=/var/lib/pgsql/passwd
```

Comment:
Just realized while trying to use this doc, that this should be 
NSS_WRAPPER_PASSWD=${HOME}/passwd